src/object.c

```diff
@@ -1246,27 +1246,23 @@ struct redisMemOverhead *getMemoryOverheadData(void) {
 
     for (j = 0; j < server.dbnum; j++) {
         redisDb *db = server.db+j;
-        long long keyscount = dictSize(db->dict);
-        if (keyscount==0) continue;
+        unsigned long long keyscount = dbSize(db, DB_MAIN);
+        if (keyscount == 0) continue;
 
         mh->total_keys += keyscount;
         mh->db = zrealloc(mh->db,sizeof(mh->db[0])*(mh->num_dbs+1));
         mh->db[mh->num_dbs].dbid = j;
 
-        mem = dictMemUsage(db->dict) +
-              dictSize(db->dict) * sizeof(robj);
+        mem = dbMemUsage(db, DB_MAIN);
         mh->db[mh->num_dbs].overhead_ht_main = mem;
         mem_total+=mem;
 
-        mem = dictMemUsage(db->expires);
-        mh->db[mh->num_dbs].overhead_ht_expires = mem;
-        mem_total+=mem;
-
-        /* Account for the slot to keys map in cluster mode */
-        mem = dictSize(db->dict) * dictEntryMetadataSize(db->dict) +
-              dictMetadataSize(db->dict);
-        mh->db[mh->num_dbs].overhead_ht_slot_to_keys = mem;
-        mem_total+=mem;
+        unsigned long long expire_keys_count = dbSize(db, DB_EXPIRES);
+        if (expire_keys_count) {       
+            mem = dbMemUsage(db, DB_EXPIRES);
+            mh->db[mh->num_dbs].overhead_ht_expires = mem;
+            mem_total+=mem;
+        }
```

Comment:
We're getting a valgrind error because `mh->db[mh->num_dbs].overhead_ht_expires` isn't being set when there are no expires. Is there a reason we aren't unconditionally setting this?

```suggestion
        mem = dbMemUsage(db, DB_EXPIRES);
        mh->db[mh->num_dbs].overhead_ht_expires = mem;
        mem_total+=mem;
```